### PR TITLE
Fixes for infra, to run build.cmd using dotnet build

### PR DESCRIPTION
**Pull request by kant2002**
This changes did not take into account that ildasm/ilasm do not support signing and windows resource embedding, for now I disable these things, so I can continue working on build infra.

/cc @leppie

### CODE REVIEW

File: IronScheme/IronScheme.Tests/IronScheme.Tests.csproj [hunk 1]
```diff
@@ -18,9 +18,6 @@
     <CodeAnalysisRuleSet />
   </PropertyGroup>
   <ItemGroup>
-    <Reference Include="nunit.framework, Version=2.6.4.14350, Culture=neutral, PublicKeyToken=96d09a1eb7f44a77, processorArchitecture=MSIL">
-      <HintPath>..\tools\nunit.framework.dll</HintPath>
-      <SpecificVersion>False</SpecificVersion>
-    </Reference>
+    <PackageReference Include="nunit.framework" Version="2.6.4" />
```
Comment (leppie) [inline IronScheme/IronScheme.Tests/IronScheme.Tests.csproj:21]: Seems we need a nuget restore before running tests.

**Review by kant2002 — COMMENTED** [review]
> Seems we need a nuget restore before running tests. 
I did revert these changes temporarily. I would like at least this land, so I can find other small subset of changes towards dotnet build usage.

File: README.md [hunk 2]
```diff
@@ -24,7 +24,7 @@ Running on Windows
 
 Run the `IronScheme.Console.exe` for the runtime and bitness you want to target. 
 
-V2 requires .NET3.5. V4 requires requires .NET4.0 or higher. Pretty much any running Windows should have one or both of those.
+V4 requires requires .NET Framework 4.7.2 or higher. Pretty much any running Windows should have this version.
```
[the remark this answers is not in the document]
Comment (kant2002) [reply README.md:27]: Let's say I can promise that I will upgrade to .NET 4.7 in near future, or I can undo this change in the file.

[the remark this answers is not in the document]
Comment (leppie) [reply README.md:27]: > I can undo this change in the file.

I prefer that.

[the remark this answers is not in the document]
Comment (kant2002) [reply README.md]: Done!

**Comment by leppie** [comment]
WIll try get to this in this week.

**Comment by leppie** [comment]
How am I supposed to test this?

I tried using `dotnet build` .

With `UseNetFxIlTools=true`  it  gets the path wrong ` "v2.0.50727\ilasm" `. (exit code 3)
With `UseNetFxIlTools=false` it  gets the path correct, but fails with exit code 1. 

I dont know how to see console output from the build process, so I dont know what is going on.

Just calling `msbuild`  works fine still.

**Comment by leppie** [comment]
Seems this has broken my local build when switching back to master.

**Comment by kant2002** [comment]
Okay. Let me explain what's goin on with `UseNetFxIlTools`
For now this change let `net20` intact and introduce this switch as a means to test non NetFX tools. I did test them by changing `build.cmd` with `dotnet build IronSchemeCore.sln /v:m /p:Configuration=%CFG% /p:UseNetFxIlTools=false`,

Overall build.cmd working (with one trivial change which I forget to commit) but `test.cmd` fails.

I did troubleshooting, and it is related to `InternalVisibleTo` and `PublicKey` on the assemblies. Actually maybe two issues
- one is that ilasm/ildasm cannot produce assemblies for runtime 2.0 or I should do some patching.
- ironscheme.boot.dll has dependency on signed variants of `IronScheme`, `IronScheme.Closure` and `IronScheme.Scripting`. I do not find sources for ironscheme.boot.dll, so I do not know how to change build process for it.

Please consider this change as a way to unlock bulding with dotnet in pricinple, even if resulting bits not yet. working. I want to move toward that goal, but I prefer to move really slow, so you can see why things should be "radically" changing here and there

**Comment by leppie** [comment]
Thanks for the feedback. I will test later.

`ironscheme.boot.dll` is the bootstrapped Scheme code for the compiler. The code is mostly in the `build` and `psyntax` folders.

It can be rebuilt by running `(ironscheme-build)` while running on a .NET desktop framework version.

The problem still remains though that it will reference signed assemblies. Now that should be easy to fix, except due to bootstrapping, there is a cyclic dependency which makes this job hard. 

I might have an idea how to fix this. The signing is probably only needed for the release package.

**Comment by leppie** [comment]
Hi,  I removed the strong naming.

Please fix that bit, and also remove the framework change to `IronScheme.Tests.Extensions` and that `<!-- NETCoreSdkRuntimeIdentifier = win-x64 -->` comment in the project file.

File: IronScheme/IronScheme.Closures/IronScheme.Closures.csproj [hunk 3]
```diff
@@ -21,13 +22,24 @@
...
+    <PropertyGroup Condition="$(UseNetFxIlTools) == True">
+      <ILAsmPath>$(TargetFrameworkSDKToolsDirectory)ilasm</ILAsmPath>
+      <ILDAsmPath>$(TargetFrameworkSDKToolsDirectory)ildasm</ILDAsmPath>
+    </PropertyGroup>
+    <PropertyGroup Condition="$(UseNetFxIlTools) != True">
+      <ILAsmPath>$(Pkgruntime_win-x64_Microsoft_NETCore_ILAsm)\runtimes\win-x64\native\ilasm</ILAsmPath>
+      <ILDAsmPath>$(Pkgruntime_win-x64_Microsoft_NETCore_ILDAsm)\runtimes\win-x64\native\ildasm</ILDAsmPath>
+    </PropertyGroup>
+    <Exec Condition="$(UseNetFxIlTools) == True" Command="&quot;$(TargetFrameworkSDKToolsDirectory)ildasm&quot; /out:&quot;$(TargetDir)$(TargetName).il&quot; /NOBAR /LINENUM &quot;$(TargetPath)&quot;&#xD;&#xA;&quot;$(SolutionDir)TailCallMaker\$(OutDir)TailCallMaker.exe&quot; &quot;$(TargetDir)$(TargetName).il&quot;&#xD;&#xA;&quot;$(TargetedFrameworkDir)\ilasm&quot; /dll /PDB /NOLOGO /QUIET /out:&quot;$(TargetDir)$(TargetName).dll&quot; /res:&quot;$(TargetDir)$(TargetName).res&quot; &quot;$(TargetDir)$(TargetName).il&quot; &quot;$(TargetDir)IronScheme.Glue.il&quot;" />
```
Comment (kant2002) [inline IronScheme/IronScheme.Closures/IronScheme.Closures.csproj:42]: All that magic is due to fact that `/res` is also don't supported by newer ILDasm.

**Comment by kant2002** [comment]
I hope that's it @leppie

**Comment by leppie** [comment]
Thanks! Hope this gets  you to where you going :D 

Let me know if you have other blockers like strong naming. 

I think that resource file can probably be ignored (might evven work for net2 target, submit PR if it does).

**Comment by leppie** [comment]
https://github.com/IronScheme/IronScheme/releases/tag/1.0.343

**Pull request merged**